SpaceDome/src/player.cpp

```diff
@@ -82,23 +86,43 @@ int Player::update(const std::vector<std::unique_ptr<Bullet>>& mBullets, float h
         return -1; 
     }
 
-    //check if hit by an enemy team bullet, if so temporary disable
+    // om spelaren har åkt på en booster
+    if(boosters.size() != 0){
+        for (auto it = boosters.begin(); it != boosters.end();) {
+            //avaktiverar en booster
+             (*it)->update(*this);
+             
+            // för att inte ta upp onödigt minne tas boostern bort från vektorn efter att den avaktiveras
+            if ((*it)->isDeactivated()) {
+                it = boosters.erase(it); 
+            } else {
+                ++it;
+            }
+        }
+    }
+
     for (const auto& bullet : mBullets) {
         if (bullet->getTeam() != mTeam) {
             glm::vec3 bulletPos = bullet->getPosition();
             float distance = glm::distance(bulletPos, mPosition);
-
-            if (distance <= hitRadius) {
-                mIsAlive = false;
-                dropStars = true; 
-                std::cout << "Player with ID: " << mPlayerID << " Was Eliminated. \n";
-                return bullet->getID();
+                if (distance <= hitRadius) {
+                    // om spelaren inte har en sköljd ska den dö direkt, annars ska den avaktivera sköljden vilket görs genom shieldHit
+                    if(mHasShield){
+                        for(auto& booster : boosters){
+                            if (auto shieldBooster = dynamic_cast<ShieldBooster*>(booster.get())){
```

Comment:
Sköldhantering bör göras i `booster::update()` så att du inte behöver lägga in specialkod för varje typ av booster.